src/browser/remote_subscriptions.ts

```diff
@@ -0,0 +1,240 @@
+/*
+Copyright 2017 OpenFin Inc.
+
+Licensed under OpenFin Commercial License you may not use this file except in compliance with your Commercial License.
+Please contact OpenFin Inc. at sales@openfin.co to obtain a Commercial License.
+*/
+
+/**
+ * DESCRIPTION
+ *
+ * This module handles remote subscriptions (multi-runtime).
+ *
+ * When adding a remote subscription, it will first check if the identity you are subscribing to exists in
+ * another runtime that is already connected.
+ *
+ * If the identity is not found, it will store the subscription locally in the list of pending subscription
+ * and will also apply the subscription to all the runtimes that are already connected.
+ *
+ * Once the listener is fired from one of the connected runtimes, it will clean the subscription from other
+ * connected runtimes and will remove the subscription from the list of pending subscriptions. In addition
+ * to doing the cleanup, it will persist the subscription to handle the case where the runtime exits and
+ * then may come back up again in the future.
+ */
+
+import * as _ from 'underscore';
+import ofEvents from './of_events';
+import connectionManager, { PeerRuntime } from './connection_manager';
+
+// id count to generate IDs for subscriptions
+let subscriptionIdCount = 0;
+// all pending remote subscriptions are stored here
+const pendingRemoteSubscriptions: Map<number, IRemoteSubscription> = new Map();
+
+/**
+ * Shape of remote subscription props
+ */
+interface IRemoteSubscriptionProps {
+    uuid: string; // app uuid
+    name?: string; // window name
+    className: 'application'|'window'; // names of the class event emitters, used for subscriptions
```

Comment:
@rdepena For external-application part, is [this](https://github.com/HadoukenIO/core/pull/20/files#diff-12aad93b38def9dae5259fe1abb6029bL183) handling it fully or not?